include/libtorrent/file_storage.hpp

```diff
@@ -559,13 +559,15 @@ TORRENT_VERSION_NAMESPACE_4
 		// returns the index of the file at the given offset in the torrent
 		file_index_t file_index_at_offset(std::int64_t offset) const;
 		file_index_t file_index_at_piece(piece_index_t piece) const;
+		file_index_t last_file_index_at_piece(piece_index_t piece) const;
 
 		// finds the file with the given root hash and returns its index
 		// if there is no file with the root hash, file_index_t{-1} is returned
 		file_index_t file_index_for_root(sha256_hash const& root_hash) const;
 
 		// returns the piece index the given file starts at
 		piece_index_t piece_index_at_file(file_index_t f) const;
+		piece_index_t last_piece_index_at_file(file_index_t f) const;
```

Comment:
ideally there would be a unit test for these two new functions, to make sure all the edge cases are handled correctly.

Reply:
That's what I thought at first. But looking at the existing unit tests for `file_storage`, you do not feel the need to test the simple logic of one line functions, for example the two existing sibling functions to which I added these two functions to don't have unit tests. However you did have a unit test for the ten line offset function called by two of these functions, which makes sense to me.